### ISSUE TITLE
VPA does not provide recommendations for resources with a parent resource that defines `/scale` subresource

### ISSUE THREAD

**Issue by shreyas-s-rao** [gardener/etcd-druid#560]
**What happened**:
VPA does not provide recommendations for resources whose parent resource defines the `/scale` subresource.
Case in point: the etcd statefulset in gardener is targeted by the VPA in order to provide scaling recommendations for the etcd pod(s). The etcd statefulset is controlled by etcd-druid, which generally did not specify the `ownerRef` on it. With g/g v1.53.0, etcd-druid started setting `ownerRef` on the etcd statefulset, which pointed to the parent `Etcd` resource. The `Etcd` CRD defines the `/scale` subresource, for the purpose of scaling the statefulset via the `Etcd` resource. Due to this change, VPA stopped providing recommendations to the etcd statefulset.
The issue was identified by @istvanballok, who also verified that the issue was resolved when either the `ownerRef` was removed, or the `/scale` subresource was removed from the `Etcd` CRD.

**What you expected to happen**:
VPA should have provided recommendations. The rationale behind the decision by VPA to not provide recommendations during such a situation is probably to avoid two controllers modifying the same field `spec.resources`. But since we use the VPA in `updateMode: Off`, the recommendations should have been provided anyway.

**How to reproduce it (as minimally and precisely as possible)**:
Run gardener `v1.53.1`. You will observe that VPA does not provide new recommendations for `etcd-main` and `etcd-events` VPA objects in the control plane of the shoot cluster. 

/cc @andrerun @istvanballok @unmarshall @timuthy 

Temporarily fixed in https://github.com/gardener/gardener/pull/6850 by removing `/scale` subresource from the `Etcd` CRD, but requires a permanent fix from VPA if we are to re-enable `/scale` subresource for `Etcd` in gardener.

**Comment by ialidzhikov** [comment]
I agree with https://github.com/gardener/etcd-druid/issues/560#issuecomment-1476133265.

> [!NOTE]  
> If the Etcd resource defines a `/scale` subresource, then the etcd VPA should target the Etcd resource.
> If the Etcd resource does NOT define a `/scale` subresource, then the etcd VPA should target the etcd StatefulSet.

@voelzmo shared today that a `/scale` subresource can not only be used for fetching the target selector but also to modify the replicas count. And that the @gardener/etcd-druid-maintainers didn't want to have a `/scale` subresource because it is not expected to be able to modify the Etcd resource replica count via the subresource (?).

Hence, I don't see something to fix/improve in the upstream VPA. If you decide to introduce again a `/scale` subresource, then VPA should target the Etcd resource (not the StatefulSet). Let us know what are your plans wrt to the `/scale` subresource.

If you don't plan to introduce a `/scale` subresource again, we can close this issue.

**Comment by unmarshall** [comment]
@voelzmo, @ialidzhikov  we (@shreyas-s-rao  and I) tested this out locally by doing the following:
* Reintroduced `scale` subresource for Etcd resource.
* Added a `Etcd.Status.Selector` and populated it correctly.
* Created a VPA resource and set its `spec.TargetRef` to the etcd resource.
* Switched off auto mode for etcd-backup-restore container and set auto for `etcd` main container. This is also how g/g configures it today.

We were able to see that VPA is scaling pods identified by `etcd.status.selector`, applying the recommendations directly on the pods of an etcd cluster.

>@voelzmo shared today that a /scale subresource can not only be used for fetching the target selector but also to modify the replicas count.

We also discussed this and today even if someone uses `kubectl scale` to increase the replicas for an `Etcd` resource, then  it will come into effect only when auto-reconciliation is enabled. The default mode (which g/g also uses) is to only reconcile an `Etcd` resource if a reconcile annotation is added to the `Etcd` resource. If the defaults are used the `kubectl scale` will not come into effect. I think this is not an issue as we can clearly document this.

So the conclusion is that we can enable `scale` subresource for `Etcd` resource, instead of doing this on StatefulSet.
I have now created https://github.com/gardener/etcd-druid/issues/1069
